joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -63,7 +63,8 @@ trait AstForTypesCreator(implicit withSchemaValidation: ValidationMode) { this:
       */
     def populateAstParentValues(typeDecl: NewTypeDecl, astParentFullName: String): NewTypeDecl = {
       val namespaceBlockFullName = s"${scope.surroundingScopeFullName.getOrElse("")}.$astParentFullName"
-      scope.pushNewScope(NamespaceScope(s"${scope.surroundingScopeFullName.getOrElse("")}.$astParentFullName"))
+      scope.pushNewScope(NamespaceScope(namespaceBlockFullName))
+      pushAccessModifier(ModifierTypes.PUBLIC)
```

Comment:
Why do we assume that the modfiier here is `PUBLIC`?

Reply:
Hmm, seems wrong to me, I think I meant to place that higher